formidable/forms/__init__.py

```diff
@@ -50,10 +50,27 @@ def __init__(self, *args, **kwargs):
         super(BaseDynamicForm, self).__init__(*args, **kwargs)
         self._bound_fields_cache = FormidableBoundFieldCache()
 
+    def rule_has_empty_fields(self, rule, data):
```

Comment:
sometimes you don't want the rule to be aware for the form/fields and sometimes you want :)
I can move that code in the Presets class but it will change the run method signature